modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/consensus/HealthCheckConsensus.scala

```diff
@@ -52,12 +52,38 @@ abstract class HealthCheckConsensus[
 
   def onOutcome(outcomes: ConsensusRounds.Outcome[F, K, A, B, C]): F[Unit]
 
+  def requestProposal(peer: PeerId, round: HealthCheckRoundId): F[Option[B]]
+
   def roundsInProgress: F[InProgress[F, K, A, B, C]] = allRounds.get.map(_.inProgress)
-  def historicalRounds: F[List[HistoricalRound[K]]] = allRounds.get.map(_.historical)
+  def historicalRounds: F[List[HistoricalRound[K, A, B]]] = allRounds.get.map(_.historical)
 
   def peersUnderConsensus: F[Set[PeerId]] =
     roundsInProgress.map(_.keySet.map(_.id))
 
+  def getOwnProposal(roundId: HealthCheckRoundId): F[Option[B]] = {
+    def inProgress =
+      roundsInProgress.flatMap { r =>
+        r.values.toList.traverse { round =>
+          round.getRoundIds.map { ids =>
+            (round, ids)
+          }
+        }
+      }.flatMap { r =>
+        r.find {
+          case (_, ids) => ids.contains(roundId)
+        }.map(_._1).traverse(_.ownConsensusHealthStatus)
+      }
+
+    def historical = historicalRounds.map { r =>
+      r.find(_.roundIds.contains(roundId)).map(_.ownProposal)
+    }
+
+    historical.flatMap {
+      case Some(proposal) => proposal.some.pure[F]
+      case _              => inProgress
+    }
```

Comment:
`82` is just an identity. We could potentially use `OptionT` because you actually need `orElseF` there:

```scala
OptionT(historical).orElseF(inProgress).value
```